src/roiextractors/extractors/memmapextractor/memmapextractors.py

```diff
@@ -0,0 +1,136 @@
+import os
+import shutil
+import tempfile
+from pathlib import Path
+from typing import Tuple, Dict
+
+import numpy as np
+from tqdm import tqdm
+
+from ...imagingextractor import ImagingExtractor
+from typing import Tuple, Dict
+
+from ...extraction_tools import (
+    PathType,
+    DtypeType,
+)
+
+
+class MemmapImagingExtractor(ImagingExtractor):
+
+    extractor_name = "MemmapImagingExtractor"
+
+    def __init__(
```

Comment:
why not just include the memmap here instead of the filename? Then this would work for HDF5 too

Reply:
Basing design off of the SpikeInterface analog: https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/binaryrecordingextractor.py

Also, part of the work here being done by the `__init__` is figuring out 'how' to read said memmap, which might be a tad much to ask first-time users.

Also also, we already have HDF5 Imaging handled: https://github.com/catalystneuro/roiextractors/blob/master/src/roiextractors/extractors/hdf5imagingextractor/hdf5imagingextractor.py, unless you meant refactoring using a kind of base binary-like class for both use cases?

Reply:
Actually, that's not a bad idea...

@h-mayorquin What do you think about this object hiearchy? `BaseImaging` -> `MemmapImaging` (which can also then handle HDF5 datasets, and any general sliceable memory map, not just numpy ones) which can then splinter into both (1) -> `HDF5Imaging` and (2) -> `BinaryImaging`.

Reply:
I don't know enough `Hdf5` to have an idea of what is to be unified. What will be the common core that is shared in `MemmapImaging`. Looking at the `Hdf5ImagingExtractor` it seems that it does file loading assuming a very specific structure of the hdf5 file. Is the reading that will be unified? the writing? the structure that the videos have inside the class?

Reply:
The `_video` object at this line: https://github.com/catalystneuro/roiextractors/blob/master/src/roiextractors/extractors/hdf5imagingextractor/hdf5imagingextractor.py#L79

is a sliceable memmap-like array whose operations largely (up to transpose, which is why you need `lazy_ops`) match `np.memmap`

The idea is, the `MemmapImaing(memmap=my_memmap_object)` would define the `get_frames()` method, and anything else that makes sense given what you know about the `memmap` object, like it's shape.

The bottom class `__init__` methods would handle all of the format-specific I/O stuff and then hand off their internal objects to the more general extractor.

Look into the structure of the old NeuroscopeRecordingExtractor with respect to the BinDatRecordingExtractor to see what I mean about this kind of structure.

Reply:
Thanks, @CodyCBakerPhD . That does clarify, the handling of the video inside both classes is the thing to be shared. I will check the example you mention.